sqlmesh/core/snapshot/definition.py

```diff
@@ -690,6 +694,12 @@ def hydrate_with_intervals_by_version(
             )
             for interval in snapshot_intervals:
                 snapshot.merge_intervals(interval)
+
+                if interval.last_altered_ts:
```

Comment:
I don't think this is correct. I don't see us distinguishing between dev and non-dev intervals. For example, adding an interval for a dev table should not have impact on `last_altered_ts` used for the non-dev / prod table.